### ISSUE TITLE
Container block selection may be lost when window regains focus

### ISSUE THREAD

**Issue by stokesman** [WordPress/gutenberg#64617]
### Description

If a container block with multiple inner blocks is selected by way of the "Select all" shortcut and the window loses focus when the window regains focus the first inner block will be selected unexpectedly. The expected behavior is that the container block selection is maintained.

It’s notable that the bug doesn’t present in the following scenarios:
- If the container block is selected by clicking the "Select parent block" toolbar button or with the breadcrumbs
- If the container block only has a single inner block

### Step-by-step reproduction instructions

1. Using the following markup
```html
<!-- wp:group {"layout":{"type":"constrained"}} -->
<div class="wp-block-group"><!-- wp:paragraph -->
<p>one</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>two</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->
```
2. Select one of the Paragraph blocks
3. Use the "Select all" shortcut until the Group block is selected
4. Switch to a different browser tab or separate window and then switch back
5. Note the selected block has changed

### Screenshots, screen recording, code snippet

https://github.com/user-attachments/assets/5361d30f-22c1-4b9f-b058-d8d9225e30e1

### Environment info

- Gutenberg Trunk
- Chrome
- MacOS

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by stokesman** [comment]
I suspect this might have the same root cause as #58788.

**Comment by stokesman** [comment]
This ended up resolved by #63671.
